lib/datadog/statsd/message_buffer.rb

```diff
@@ -83,6 +95,21 @@ def ensure_sendable!(message_size)
       def bytesize_threshold
         @bytesize_threshold ||= (max_payload_size - PAYLOAD_SIZE_TOLERANCE * max_payload_size).to_i
       end
+
+      # below are "fork management" methods to be able to clean the MessageBuffer
+      # if it detects that it is running in a unknown PID.
+
+      def forked?
+        Process.pid != fork_pid
+      end
+
+      def update_fork_pid
+        @fork_pid = Process.pid
+      end
+
+      def fork_pid
+        @fork_pid ||= Process.pid
+      end
```

Comment:
Minor: Since `#update_fork_pid` is being added to the right places, I don't think the `||=` is needed on `#fork_pid`, so I suggest removing it to avoid future bugs "oops I didn't call `#update_fork_pid` and nothing broke".

Reply:
100% agreeing, removing this. Note that I've re-used that snippet [from here](https://github.com/DataDog/dd-trace-rb/blob/e391d2eb64d3c6151a4bdd2710c6a8c7c1d57457/lib/ddtrace/utils/forking.rb), however, I always prefer explicit against implicit.

Reply:
😅  Hehe yeah there's a few bits in dd-trace-rb that I would like to change when I get the opportunity :)